### ISSUE TITLE
Custom fields by group error

### ISSUE THREAD

**Issue by beechyrb** [opencart/opencart#3355]
Custom fields are being retrieved by default config customer group instead of logged in customer group in the account edit controller.

catalog/controller/account/edit.php

Line 159
$data['custom_fields'] = $this->model_account_custom_field->getCustomFields($this->config->get('config_customer_group_id'));

**Fix**
$data['custom_fields'] = $this->model_account_custom_field->getCustomFields($this->customer->getGroupId());

Line 209
$data['custom_fields'] = $this->model_account_custom_field->getCustomFields($this->config->get('config_customer_group_id'));

**Fix**
$custom_fields = $this->model_account_custom_field->getCustomFields($this->customer->getGroupId());

**Comment by GomDani** [comment]
This is set in index.php

```
if ($customer->isLogged()) {
	$config->set('config_customer_group_id', $customer->getGroupId());
}
```

**Comment by beechyrb** [comment]
Oops I see that now. I should have output the results first.
It seems like a sort issue for my custom field then, if I set it to 6 it shows on the edit page but not on the register page, if I set it to 5 it shows on the register page but not the edit. Any ideas? I will keep looking for the issue.

**Comment by danielkerr** [comment]
i have no idea which version you are using?

**Comment by beechyrb** [comment]
My apologies I seems to always forget the version. I am on 2.0.3.1

The issue is with the jquery in catalog/view/theme/default/template/account/register.tpl starting on line 533

$('#account .form-group[data-sort]').detach().each(function() {
	if ($(this).attr('data-sort') >= 0 && $(this).attr('data-sort') <= $('#account .form-group').length) {
		$('#account .form-group').eq($(this).attr('data-sort')).before(this);
	}

	if ($(this).attr('data-sort') > $('#account .form-group').length) {
		$('#account .form-group:last').after(this);
	}

	if ($(this).attr('data-sort') < -$('#account .form-group').length) {
		$('#account .form-group:first').before(this);
	}
});

Unfortunately I am no jquery expert so i have no idea why it **detach** if you set the sort to 6. I wanted my custom field at the end so my method to overcome this was to set the sort to 7 which works fine on both the register and edit account pages. I do not allow the customer to select which customer group to join upon registration so that hidden customer group field is what seems to be interfering.

**Comment by danielkerr** [comment]
i think this has been fixed in the master version

**Comment by danielkerr** [comment]
yes you need to -1 it says thisif you hover over the sort order and see the help text.
